docs/sphinx/heps/hep2.rst

```diff
@@ -0,0 +1,312 @@
+  - distributed equivalents of parallel algorithms
+
+- ``hpx/distributed/algorithm/for_loop.hpp``
+  - distributed equivalents of parallel algorithms
+
+- ``hpx/distributed/latch.hpp``
+  - ``hpx::distributed::latch``
+
+- ``hpx/distributed/barrier.hpp``
+  - ``hpx::distributed::barrier``
+
+- ``hpx/distributed/channel.hpp``
+  - ``hpx::distributed::channel``
+
+- etc. to match local headers and namespaces
+
+- ``hpx/distributed/components.hpp``
+  - ``HPX_REGISTER_COMPONENT``
+  - any other classes, functions, and macros that might be required to write components?
+
+- ``hpx/distributed/actions.hpp``
+  - ``HPX_REGISTER_ACTION``
+  - any other classes, functions, and macros that might be required to write actions?
+
+- ``hpx/distributed/x.hpp`` each in a separate header like channel, latch, etc.?
+  - ``hpx::distributed::all_to_all``
+  - ``hpx::distributed::all_reduce``
+  - ``hpx::distributed::broadcast``
+  - ``hpx::distributed::fold``
+  - ``hpx::distributed::gather``
+  - ``hpx::distributed::reduce``
+  - ``hpx::distributed::spmd_block``
+
+- ``hpx/distributed/checkpoint.hpp``?
+  - ``hpx::distributed::checkpoint`` stable?
+  - this is not necessarily distributed only...
+
+- ``hpx/experimental/distributed/resiliency.hpp``?
+  - ``hpx::experimental::distributed::async_replay`` and other resiliency?
+  - ready for public use or still experimental?
+  - the standard libraries use ``std::experimental::x``, perhaps we should just
+    follow that instead? the argument for ``hpx::x::experimental`` wasn't that
+    strong anyway...
+  - this is not necessarily distributed only...
+
+- ``hpx/distributed.hpp``
+  - all above headers
+
+Runtime
+.......
+
+- ``hpx/runtime.hpp``?
+  - ``hpx::get_os_thread_count``
+  - ``hpx::get_worker_thread_num``
+  - ``hpx::get_thread_name``
+  - other ``hpx::get_*`` functions
+  - ``hpx::register_thread``
+  - ``hpx::unregister_thread``
+  - ``hpx::register_startup_function``
+  - ``hpx::register_shutdown_function``
+  - ``hpx::is_*``
+  - ``hpx::state`` (rename to ``hpx::runtime_state`` for the runtime?)
+  - ``hpx::get_locality_id`` this and the following are not quite right here,
+    but they can return reasonable defaults even in the local case...
+  - ``hpx::get_locality_name``
+  - ``hpx::get_num_localities``
+  - note: ``hpx::runtime`` does not need to be public; the above global
+    functions should be enough for interacting with the runtime. They also allow
+    us to have checks that the runtime is created or running (as we already do).
+
+- ``hpx/distributed/runtime.hpp``?
+  - all of the above and (should they be in ``hpx::distributed`` or not?)
+  - ``hpx::find_localities``
+  - ``hpx::find_here``
+
+- ``hpx/init.hpp``?
+  - ``hpx::init`` (rename to ``hpx::start_stop`` or something similar? only
+    leave ``hpx::start`` and ``hpx::stop``? I found the relationship between
+    ``start``, ``stop``, ``init``, and ``finalize`` confusing at least in the
+    beginning.)
```

Comment:
`start`/`stop` and `init` are serving different use cases. I think we should keep them. Let's improve the docs if necessary instead.

Reply:
I see the use case, it's more about the naming. I don't have better suggestions though, so we can leave them as they are.